org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/RunnableDeviceProxy.java

```diff
@@ -0,0 +1,197 @@
+		delegate.preConfigure(model, scanBean, publisher);
+	}
+
+	@PostConfigure
+	public void postConfigure(Object model, ScanBean scanBean, IPublisher<?> publisher) throws ScanningException {
+		logger.trace("postConfigure({}, {}, {}) on {}", model, scanBean, publisher, getName());
+		delegate.postConfigure(model, scanBean, publisher);
+	}
+
+	@LevelStart
+	public void levelStart(LevelInformation info) throws ScanningException {
+		logger.trace("levelStart({}) on {}", info, getName());
+		delegate.levelStart(info);
+	}
+
+	@LevelEnd
+	public void levelEnd(LevelInformation info) throws ScanningException {
+		logger.trace("levelEnd({}) on {}", info, getName());
+		delegate.levelEnd(info);
+	}
+
+	@PointStart
+	public void pointStart(IPosition point) throws ScanningException {
+		logger.trace("pointStart({}) on {} stepIndex={}", point, getName(), point.getStepIndex());
+		delegate.pointStart(point);
+	}
+
+	@PointEnd
+	public void pointEnd(IPosition point) throws ScanningException {
+		logger.trace("pointEnd({}) on {} stepIndex={}", point, getName(), point.getStepIndex());
+		delegate.pointEnd(point);
+	}
+
+	@ScanStart
+	public void scanStart(ScanInformation info) throws ScanningException {
+		logger.trace("scanStart({}) on {} filePath={}", info, getName(), info.getFilePath());
+		delegate.scanStart(info);
+	}
+
+	@ScanEnd
+	public void scanEnd(ScanInformation info) throws ScanningException {
+		logger.trace("scanEnd({}) on {} filePath={}", info, getName(), info.getFilePath());
+		delegate.scanEnd(info);
+	}
+
+	@WriteComplete
+	public void writeComplete(ScanInformation info) throws ScanningException {
+		logger.trace("writeComplete({}) on {} filePath={}", info, getName(), info.getFilePath());
+		delegate.writeComplete(info);
+	}
+
+	@ScanAbort
+	public void scanAbort(ScanInformation info) throws ScanningException {
+		logger.trace("scanAbort({}) on {}", info, getName());
+
+		delegate.scanAbort(info);
+	}
+
+	@ScanFault
+	public void scanFault(ScanInformation info) throws ScanningException {
+		logger.trace("scanFault({}) on {}", info, getName());
+		delegate.scanFault(info);
+	}
+
+	@ScanFinally
+	public void scanFinally(ScanInformation info) throws ScanningException {
+		logger.trace("scanFinally({}) on {}", info, getName());
+
+		delegate.scanFinally(info);
+	}
+
+	@ScanPause
+	public void scanPaused() throws ScanningException {
+		logger.trace("scanPaused() on {}", getName());
+
+		delegate.scanPause();
+	}
+
+	@ScanResume
+	public void scanResumed() throws ScanningException {
+		logger.trace("scanResumed() on {}", getName());
+
+		delegate.scanResume();
+	}
+
+	// Class methods
+
+	public AbstractRunnableDeviceDelegate getDelegate() {
+		return delegate;
+	}
+
+	public void setDelegate(AbstractRunnableDeviceDelegate delegate) {
+		logger.trace("scanResume({}) on {}", delegate, getName());
+
+		this.delegate = delegate;
+		if (delegate.getRunnableDeviceProxy() != this ) {
+			throw new RuntimeException("Delegates runnable device is not this!");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)